src/main.rs

```diff
@@ -60,33 +154,65 @@ fn main() {
     let parse_tree = parser::parse_prog(&mut token_iter);
 
     let mut code_gen = code_gen::CodeGen::new_with_config(config);
-    // Get the full original command.
-    code_gen.cmd = std::env::args().collect::<Vec<String>>().join(" ");
+     code_gen.cmd = std::env::args().collect::<Vec<String>>().join(" ");
 
-    // TODO(taegyunkim): Specify the root service name from commandline
-    code_gen.root_id = "productpage-v1";
-    code_gen.visit_prog(&parse_tree);
+     code_gen.root_id = matches.value_of("root_node").unwrap();
+     code_gen.visit_prog(&parse_tree);
 
-    // Open filter template.
-    let template_path = bin_dir.join("filter.cc.handlebars");
-    let display = template_path.display();
-    let mut template_file = match File::open(&template_path) {
-        Err(msg) => panic!("Failed to open {}: {}", display, msg),
-        Ok(file) => file,
-    };
 
-    let mut template_str = String::new();
-    match template_file.read_to_string(&mut template_str) {
-        Err(msg) => panic!("Failed to read {}: {}", display, msg),
-        Ok(_) => println!("Successfully read {}", display),
-    }
 
-    let handlebars = Handlebars::new();
+    // Use the information in the code generator code_gen and format it, using handlebars
+    // to a template with all the basic filter information enclosed
+    let c_mode = matches.value_of("compilation_mode").unwrap();
+    let filter_name = matches.value_of("filter_name").unwrap();
+    if c_mode == "sim" {
+        // Because we are making a library, not just one file, we need to copy over an example library.  Then,
```

Comment:
Why can we not have a default folder like with cpp_filter and just copy over the filter file?

Reply:
Both Cargo.toml and lib.rs need to know what the filter's name is.  Otherwise they cannot name what the library is.  We need to make this adjustable for the case where we have multiple filters, and they need different names to distinguish themselves.  If we have to go in and change the name in all three files every time we want to make a filter with a different name, that will get tedious.

Reply:
So ideally the simulator should behave nearly identically to the cpp style filter. There we generate a filter.cc which is put into the cpp_filter environment and is compiled there. We can do the same here. Cargo.toml and lib.rs will just depend on filter.rs, which is generated. So far there is no need for multiple filters. The file name flag is useful if we want to keep track of reference files or for other use cases, but we do not really need it here.

Reply:
Okay, I will make a rust_filter directory and only generate the filter itself then.